roles/openshift_health_checker/openshift_checks/__init__.py

```diff
@@ -107,30 +107,27 @@ def get_var(self, *keys, **kwargs):
                 return kwargs["default"]
             raise OpenShiftCheckException("'{}' is undefined".format(".".join(map(str, keys))))
 
-        convert = kwargs.get("type")
+        convert = kwargs.get("convert", lambda x: x)
         try:
-            if convert is None:
-                pass
-            elif convert in ["str", str]:
-                return str(value)
-            elif convert in ["int", int]:
-                return int(value)
-            elif convert in ["float", float]:
-                return float(value)
-            elif convert in ["bool", bool]:
+            if convert is bool:
                 return str(value).lower() in ["true", "yes", "1"]
             else:
-                raise OpenShiftCheckException(
-                    'There is a bug in this check. '
-                    'It is trying to convert a variable to unknown type "{}"'.format(convert)
-                )
+                return convert(value)
+
         except ValueError as error:
             raise OpenShiftCheckException(
-                'Cannot convert variable "{var}" to expected type "{type}":\n'
-                '{error}'.format(var=".".join(keys), type=convert, error=str(error))
+                'Cannot convert inventory variable to expected type:\n'
+                '  "{var}={value}"\n'
+                '{error}'.format(var=".".join(keys), value=value, error=str(error))
             )
 
-        return value
+        except Exception as error:
+            raise OpenShiftCheckException(
+                'There is a bug in this check. While trying to convert variable \n'
+                '  "{var}={value}"\n'
+                'the given converter cannot be used or failed unexpectedly:\n'
+                '{error}'.format(var=".".join(keys), value=value, error=error)
+            )
```

Comment:
`ValueError` ought to mean the user passed in a wrong value.

Any other kind of error indicates the conversion function writer did not handle the value correctly or at least gracefully (`KeyError` and similar). The user might have done something wrong too, but if it's not obvious to them then they can blame our code.

Are there other error categories that should be included / called out?